File: hard/reverse-nodes-in-k-group/s.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

from typing import Optional, List

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def reverseKGroup(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        d: List[int] = list(range(k))
        nHead = head
        it = head
        j = 0
        while head != None:
             d[j%k] = head.val
             j += 1
             head = head.next
             if j%k == 0:
                  for i in range(k-1, -1, -1):
                       it.val = d[i]
                       it = it.next
        return nHead

File: hard/reverse-nodes-in-k-group/s.go
package solution

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func reverseKGroup(head *ListNode, k int) *ListNode {
	d := make([]int, k)
	nHead := head
	it := head

	j := 0
	for head != nil {
		d[j%k] = head.Val
		j++
		head = head.Next

		if j%k == 0 {
			for i := k - 1; i > -1; i-- {
				it.Val = d[i]
				it = it.Next
			}
		}
	}
	return nHead
}

type ListNode struct {
	Val  int
	Next *ListNode
}
